### PR TITLE
coprocessor: fix do_div_mod bug

### PR DESCRIPTION
**Pull request by mccxj**
for #2453

### CODE REVIEW

File: src/coprocessor/codec/mysql/decimal.rs [hunk 1]
```diff
@@ -591,7 +591,7 @@ fn do_div_mod(
     let mut buf = vec![0; l_len];
     (&mut buf[0..i]).copy_from_slice(&lhs.word_buf[l_idx..l_idx + i]);
     let mut l_idx = 0;
-    let (r_start, mut r_stop) = (r_idx, word_cnt!(r_idx + r_prec as usize, usize) - 1);
+    let (r_start, mut r_stop) = (r_idx, r_idx + word_cnt!(r_prec as usize, usize) - 1);
```
Comment (BusyJay) [inline src/coprocessor/codec/mysql/decimal.rs]: Any tests to cover it?

**Comment by coocood** [comment]
@mccxj 
You can add test in `decimal.rs:2854`.

File: src/coprocessor/codec/mysql/decimal.rs (hunk 1 above)
Comment (mccxj) [reply src/coprocessor/codec/mysql/decimal.rs]: @BusyJay I'm adding test cases, please wait a minute.

**Comment by coocood** [comment]
LGTM

**Comment by BusyJay** [comment]
Thanks!

**Pull request merged**